internal/system/agent/services.go

```diff
@@ -131,6 +136,49 @@ func getMetrics(services []string, ctx context.Context) (MetricsRespMap, error)
 	return m, nil
 }
 
+func getHealth(services []string, ctx context.Context) HealthRespMap {
+
+	h := HealthRespMap{}
+	h.Health = map[string]interface{}{}
```

Comment:
Per my comment above, this could be
`health := map[string]interface{}{}`

`health` can then be returned along with the error, if any, as Akram points out below. When this is then serialized into JSON as the HTTP response, the consumer won't have the extra 'Health' node in the JSON to parse (this applies to the existing Config and Metrics endpoints as well). In my opinion, including `Health` or `Metrics` or `Config` in the response is redundant because that's already indicated by the path of the REST request. It will also require case based parsing on the consumer side.

Having said all of that, if you still want to have a type that wraps the response data (which is a [legit pattern](https://martinfowler.com/eaaCatalog/dataTransferObject.html)), you can do this with just a single type instead of three.